src/transformers/models/whisper/generation_whisper.py

```diff
@@ -0,0 +1,1436 @@
+            return_token_timestamps=return_token_timestamps, generation_config=generation_config, kwargs=kwargs
+        )
+
+        # 4. Retrieve logits processors
+        # => TODO(Patrick): The way `num_start_tokens` is retrieved here is too brittle. Need a better approach
+        num_start_tokens = (
+            len(generation_config.forced_decoder_ids) if generation_config.forced_decoder_ids is not None else 1
+        )
+        logits_processor = self._retrieve_logit_processors(
+            generation_config=generation_config,
+            logits_processor=logits_processor,
+            no_speech_threshold=no_speech_threshold,
+            num_start_tokens=num_start_tokens,
+            is_shortform=is_shortform,
+            num_beams=kwargs.get("num_beams", 1),
+        )
+
+        # 5. If we're in shortform mode, simple generate the whole input at once and return the output
+        if is_shortform:
+            if temperature is not None:
+                kwargs["temperature"] = temperature
+
+            outputs = super().generate(
+                input_features,
+                generation_config=generation_config,
+                logits_processor=logits_processor,
+                stopping_criteria=stopping_criteria,
+                prefix_allowed_tokens_fn=prefix_allowed_tokens_fn,
+                synced_gpus=synced_gpus,
+                **kwargs,
+            )
+
+            if generation_config.return_token_timestamps and hasattr(generation_config, "alignment_heads"):
+                outputs["token_timestamps"] = self._extract_token_timestamps(
+                    outputs, generation_config.alignment_heads, num_frames=generation_config.num_frames
+                )
+
+            return outputs
+
+        # 6. Else we're in longform mode which is more complex.
+        # We need to chunk the audio input depending on when the model generates timestamp tokens
+
+        # 6.1 Set and retrieve global longform generation variables
+        self._set_condition_on_prev_tokens(
+            condition_on_prev_tokens=condition_on_prev_tokens, generation_config=generation_config
+        )
+
+        timestamp_begin = generation_config.no_timestamps_token_id + 1
+        temperatures = [temperature] if not isinstance(temperature, (list, tuple)) else temperature
+        temperature = temperatures[0]
+        batch_size = input_features.shape[0]
+
+        max_frames, seek = self._retrieve_max_frames_and_seek(
+            batch_size=batch_size, attention_mask=attention_mask, total_input_frames=total_input_frames
+        )
+        init_tokens = self._retrieve_init_tokens_from_forced_decoder_ids(generation_config=generation_config)
+
+        # 6.2 Preppare running variables, list for generation
+        cur_bsz = batch_size
+        current_segments = [[] for _ in range(batch_size)]
+        batch_idx_map = list(range(batch_size))
+        do_condition_on_prev_tokens = [condition_on_prev_tokens for _ in range(batch_size)]
```

Comment:
I might be missing something here -- shouldn't `do_condition_on_prev_tokens` be reset within the next `while` loop OR not returned by `self.generate_with_fallback()`?

As it stands, each iteration starts with `do_condition_on_prev_tokens` from the previous iteration, which might have been changed in `self.generate_with_fallback()` when the previous loop iteration used `temperature > 0.5`. As such, I think this may be having an impact on the `self._prepare_decoder_input_ids()` call, causing `generate` to NOT condition on previous inputs when it should.

Again, I might be missing something here :D

Reply:
So the way it's implemented in the original codebase is that only if the previous segment was generated with a probability of > 0.5 then the **next** segment shall not be conditioned on this segment. 

I've double-checked the implementation as is now and I think it's correctly implemented

Reply:
Gotcha, thank you for clearing it out 👍 

(and it does make sense as you explained -- if a sequence was generated with high temperature [=noisy], let's not rely too much on it)